dali/operators/reader/loader/file_loader.h

```diff
@@ -112,14 +112,32 @@ class FileLoader : public Loader<Backend, Target> {
     copy_read_data_ = dont_use_mmap_ || !mmap_reserver_.CanShareMappedData();
   }
 
-  std::enable_if_t<detail::has_data<Target>::value, void>
-  PrepareEmpty(Target &target) override {
+  template <typename T>
+  std::enable_if_t<!detail::has_data<T>::value, void>
+  PrepareEmptyImpl(T &target) {
+    assert(false);
+  }
```

Comment:
Why is this necessary? It duplicates the logic from the base loader.

Reply:
Can't use std::enable_if_t on a virtual function.

Reply:
I removed it since it was not used anywhere